src/main/java/ch/njol/skript/effects/EffForceAttack.java

```diff
@@ -31,37 +30,45 @@ public class EffForceAttack extends Effect {
 	}
 	
 	private static final boolean ATTACK_IS_SUPPORTED = Skript.methodExists(LivingEntity.class, "attack", Entity.class);
-	
-	@SuppressWarnings("null")
+
+	private Node node;
 	private Expression<LivingEntity> entities;
-	@SuppressWarnings("null")
 	private Expression<Entity> target;
-	
-	@SuppressWarnings("unchecked")
+
 	@Override
+	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		if (!ATTACK_IS_SUPPORTED) {
-			Skript.error("The force attack effect requires Minecraft 1.15.2 or newer");
+			Skript.error("The force attack effect requires Minecraft 1.15.2+");
 			return false;
 		}
+		node = getParser().getNode();
 		entities = (Expression<LivingEntity>) exprs[0];
 		target = (Expression<Entity>) exprs[1];
 		return true;
 	}
 	
 	@Override
-	protected void execute(Event e) {
-		Entity target = this.target.getSingle(e);
-		if (target != null) {
-			for (LivingEntity entity : entities.getArray(e)) {
-				entity.attack(target);
-			}
+	protected void execute(Event event) {
+		Entity target = this.target.getSingle(event);
+		if (target == null) {
+			error("The provided target entity was not set.", this.target.toString());
```

Comment:
i'm not sure this one needs to exist
making them attack nothing and nothing happening seems like good behavior